plugins/vpp/ifplugin/bfd_config.go

```diff
@@ -414,3 +428,12 @@ func (plugin *BFDConfigurator) DeleteBfdEchoFunction(bfdInput *bfd.SingleHopBFD_
 func AuthKeyIdentifier(id uint32) string {
 	return strconv.Itoa(int(id))
 }
+
+// LogError prints error if not nil, including stack trace. The same value is also returned, so it can be easily propagated further
+func (c *BFDConfigurator) LogError(err error) error {
+	if err == nil {
+		return nil
+	}
+	c.log.WithField("logger", c.log).Errorf(string(err.Error() + "\n" + string(err.(*errors.Error).Stack())))
```

Comment:
This might need some type check to verify that `err` really is `*errors.Error` type, before using `Stack` method. Otherwise sooner or later someone could call `LogError` with different type and realize it when it fails in runtime.

Reply:
Also, you might wanna use [ErrorStack()](https://godoc.org/github.com/go-errors/errors#Error.ErrorStack) here.

Reply:
I agree with the first part. But I don't want to use ErrorStack(), it prints TypeName of the error which is useless and only decreases readability